config/config_export_variables.go

```diff
@@ -0,0 +1,78 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/coveooss/gotemplate/v3/hcl"
+	"github.com/hashicorp/terraform/configs"
+	"gopkg.in/yaml.v2"
+)
+
+type ExportVariables struct {
+	Path   string `hcl:"path"`
+	Format string `hcl:"format,optional"`
+}
+
+// SaveVariables saves variables in paths defined in the export_variables blocks
+func (conf *TerragruntConfig) SaveVariables(existingTerraformVariables map[string]*configs.Variable, folders ...string) (err error) {
+	variables := conf.options.GetContext()
+
+	for _, exportStatement := range conf.ExportVariables {
+		exportStatement.Format = strings.TrimSpace(exportStatement.Format)
+		if exportStatement.Format == "" {
+			exportStatement.Format = strings.Trim(filepath.Ext(exportStatement.Path), ". ")
```

Comment:
why `. ` with a space?

Reply:
Trims both spaces and dots. The Trim function accepts a charset (string)